pkg/tempopb/tempo.proto

```diff
@@ -52,11 +52,14 @@ message TraceByIDRequest {
   string blockStart = 2;
   string blockEnd = 3;
   string queryMode = 5;
+  bool allowPartialTrace = 6;
 }
 
 message TraceByIDResponse {
   Trace trace = 1;
   TraceByIDMetrics metrics = 2;
+  bool partialTrace = 3;
+  string warning = 4;
```

Comment:
```
  bool partialTrace = 3;
  string warning = 4;
```

Perhaps an enum instead of a bool in case we want to add more functionality here in the future?

nit: `string message` instead of `string warning` ?

Reply:
Seems reasonable to me